### PR TITLE
Pl-130311 fix k3s garbagecollect sudo tests

### PR DESCRIPTION
**Pull request by elikoga**
@flyingcircusio/release-managers

## Release process

Impact:

Changelog:

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
  - existing tests are modified to un-fail 
- [x] Security requirements tested? (EVIDENCE)
  - tests run on hydra00

### CODE REVIEW

**Comment by elikoga** [comment]
will squash once reviewed

File: tests/k3s/airgapped-k3s-images.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    r = requests.get(url)
+    # split lines
+    lines = r.text.splitlines()
+    # parse lines
+    images = [re.match(image_regex, line).groupdict() for line in lines]
+    # prefetch images
+    imageNix = []
+    for image in images:
+        print(f"prefetching {image['image']}:{image['tag']}")
+        p = subprocess.run(["nix-prefetch-docker", f"{image['image']}", f"{image['tag']}"], capture_output=True)
```
Comment (dpausp) [inline tests/k3s/airgapped-k3s-images.py]: use image['image'] as argument directly, string interpolation is a no-op here

Comment (dpausp) [reply tests/k3s/airgapped-k3s-images.py]: `check=True` to catch exit codes > 0 here, `text=True` to decode utf8 (so you can append p.stdout directly in line 36)

File: tests/k3s/airgapped-k3s-images.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        # stdout contains nix code to neccessary to fetch the image
+        imageNix.append(p.stdout.decode("utf-8"))
+    # write to file
+    with open("airgapped-k3s-images.nix", "w") as f:
+        f.write(f"""[
```
Comment (dpausp) [inline tests/k3s/airgapped-k3s-images.py]: writing in 3 parts (opening bracket, content, closing bracket) looks cleaner than a multiline string here

**Review by dpausp — CHANGES_REQUESTED** [review]

**Pull request merged**